### PR TITLE
internal/lang: `ephemeralasnull`: Add test for unknown values

### PR DESCRIPTION
**Pull request by radeksimko**
This is a small PR following up on https://github.com/hashicorp/terraform/pull/35652 which I wanted to split from https://github.com/hashicorp/terraform/pull/35653 as that PR may require some further discussions.

It covers the extra logic present in the function implementation concerning unknown values:

https://github.com/hashicorp/terraform/blob/e312ffc2d821901e982f897de952944070f74f19/internal/lang/funcs/conversion.go#L132-L144

In other words, the test (and underlying logic under test) ensures that other marks (other than ephemerality) are retained while passing through that function.

### CODE REVIEW

**Review by jbardin — APPROVED** [review]

**Comment by apparentlymart** [comment]
This seems to be gradually growing to replicate several cases already covered in this function's main tests:

https://github.com/hashicorp/terraform/blob/a2039517a5300e831092cdca31b9cb44b84f0d0c/internal/lang/funcs/conversion_test.go#L210-L349

Typically these function-table-level tests have been used only to test whether a function is correctly registered in the function table so it can be used at all, rather than getting into all the weeds of how the function behaves. No significant harm in adding these additional tests of course, but I point it out in case you've been doing this just because you didn't know the other test cases were there.

(This mostly-redundant set of tests in the `lang` package came from a few instances where someone had mistakenly added a function in `lang/funcs` but had either omitted it from the functions table or had specified it incorrectly so that it couldn't actually be reached, so we added an extra layer of testing to try to cover that mistake.)

**Comment by radeksimko** [comment]
@apparentlymart Good catch! I didn't notice these tests earlier. I should've checked the test cover more thoroughly..
